plugins/rebalance/rebalance.go

```diff
@@ -0,0 +1,109 @@
+	rebalanceapi "github.com/gluster/glusterd2/plugins/rebalance/api"
+	config "github.com/spf13/viper"
+)
+
+const (
+	sbinDir = "glusterfs"
+)
+
+// Process type represents information about rebalance process
+type Process struct {
+	binarypath     string
+	args           string
+	socketfilepath string
+	pidfilepath    string
+	rebalance      rebalanceapi.RebalInfo
+}
+
+// Name returns the process name
+func (r *Process) Name() string {
+	return "rebalance"
+}
+
+// Path returns absolute path to the binary of rebalance process
+func (r *Process) Path() string {
+	return r.binarypath
+}
+
+// Args returns arguments to be passed to rebalance process
+func (r *Process) Args() string {
+
+	volfileserver, _, _ := net.SplitHostPort(config.GetString("clientaddress"))
+	if volfileserver == "" {
+		volfileserver = "localhost"
+	}
+
+	volFileID := path.Join("rebalance/", r.rebalance.Volname)
+	File := path.Join(config.GetString("logdir"), "glusterfs", r.rebalance.Volname)
+	logFile := fmt.Sprintf("%s-rebalance.log", File)
+	cmd := r.rebalance.Cmd
+	commithash := r.rebalance.CommitHash
+
+	var buffer bytes.Buffer
+	buffer.WriteString(fmt.Sprintf(" -s %s", volfileserver))
+	buffer.WriteString(fmt.Sprintf(" --volfile-id %s", volFileID))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.use-readdirp=yes"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.lookup-unhashed=yes"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.assert-no-child-down=yes"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option replicate.data-self-heal=off"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option replicate.metadata-self-heal=off"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option replicate.entry-self-heal=off"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.readdir-optimize=on"))
+	buffer.WriteString(fmt.Sprintf(" --process-name rebalance"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.rebalance-cmd=%d", cmd))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.node-uuid=%s", gdctx.MyUUID))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.commit-hash=%d", commithash))
+	buffer.WriteString(fmt.Sprintf(" -p %s", r.PidFile()))
+	buffer.WriteString(fmt.Sprintf(" -S %s", r.SocketFile()))
+	buffer.WriteString(fmt.Sprintf(" -l %s", logFile))
+	r.args = buffer.String()
+	return r.args
+}
+
+// SocketFile returns path to the socket file used for IPC
+func (r *Process) SocketFile() string {
+
+	if r.socketfilepath != "" {
+		return r.socketfilepath
+	}
+
+	glusterdSockDir := path.Join(config.GetString("rundir"), "gluster", r.rebalance.Volname)
+	r.socketfilepath = fmt.Sprintf("%s-rebalance-%x.socket", glusterdSockDir, xxhash.Sum64String(gdctx.MyUUID.String()))
+	return r.socketfilepath
+}
+
+// PidFile returns path to the pid file of rebalance process
+func (r *Process) PidFile() string {
+	if r.pidfilepath != "" {
+		return r.pidfilepath
+	}
+	filepath := path.Join(config.GetString("rundir"), "gluster", r.rebalance.Volname)
```

Comment:
nit: It's more intuitive to name this variable as `pidDir` and move `r.rebalance.Volname` to `fmt.Sprintf` call.